### PR TITLE
ODTR-23 | Scenario supported languages

### PR DESCRIPTION
**Pull request by goranambar**
Adds support for `base language` and `supported languages` for `Scenario`. Updates the data clients to store and retrieve the languages properties of the `Scenario`

### CODE REVIEW

File: src/Conversation/Scenario.php [hunk 1]
```diff
@@ -147,4 +155,44 @@ public function deactivate(): bool
...
+    {
+        $this->baseLanguage = $value;
+    }
+
+    /**
+     * Gets a list of scenario supported language codes.
+     *
+     * @return ?array
+     */
+    public function getSupportedLanguages(): ?array
```
Comment (stu-greenshoots) [inline src/Conversation/Scenario.php:184]: I guess this would always return an array as it is initialised?

File: src/CoreServiceProvider.php [hunk 2]
```diff
@@ -64,11 +68,15 @@ public function boot()
...
-                ComponentSettings::class,
-            ]);
-        }
+        $this->commands([
+            ComponentSettings::class,
+            LanguagesSeed::class,
+            ScenariosLanguageUpdate::class
+        ]);
+
+        $this->app->singleton(TranslateServiceInterface::class, function () {
```
Comment (stu-greenshoots) [inline src/CoreServiceProvider.php:77]: A small change, but could we create a TranslateServiceProvider as per the other modules? Just to keep everything together. 
Eg the ActionEngineServiceProvider

**Review by stu-greenshoots — CHANGES_REQUESTED** [review]
Thanks Goran, I tested this and it looks good to me. Just a couple of very small comments for you to look at

File: src/Conversation/Scenario.php (hunk 1 above)
Comment (goranambar) [reply src/Conversation/Scenario.php]: Hey @stu-greenshoots this is simillar as `status` and `active`, which are both `nullable` and can be set to null in  [dehydrate](https://github.com/opendialogai/core/blob/14cf399f6173ab66f3e4adcfa6b5b684fb83483d/src/Conversation/Scenario.php#L48).

**Review by stu-greenshoots — APPROVED** [review]

**Pull request merged**